sdk-crypto/src/main/java/org/fisco/bcos/sdk/crypto/keypair/SDFSM2KeyPair.java

```diff
@@ -0,0 +1,139 @@
+package org.fisco.bcos.sdk.crypto.keypair;
+
+import com.webank.blockchain.hsm.crypto.sdf.AlgorithmType;
+import com.webank.blockchain.hsm.crypto.sdf.SDF;
+import com.webank.blockchain.hsm.crypto.sdf.SDFCryptoResult;
+import com.webank.wedpr.crypto.CryptoResult;
+import java.math.BigInteger;
+import java.security.KeyPair;
+import org.fisco.bcos.sdk.crypto.exceptions.KeyPairException;
+import org.fisco.bcos.sdk.crypto.hash.Hash;
+import org.fisco.bcos.sdk.crypto.hash.SDFSM3Hash;
+import org.fisco.bcos.sdk.crypto.keystore.KeyTool;
+import org.fisco.bcos.sdk.utils.Hex;
+import org.fisco.bcos.sdk.utils.Numeric;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SDFSM2KeyPair extends CryptoKeyPair {
+    protected static Logger logger = LoggerFactory.getLogger(SDFSM2KeyPair.class);
+    public static Hash DefaultHashAlgorithm = new SDFSM3Hash();
+    public long keyIndex;
+    public String password;
+    public boolean isInternalKey = false;
+
+    public SDFSM2KeyPair() {
+        initSM2KeyPairObject();
+        CryptoKeyPair keyPair = this.generateKeyPair();
+        this.hexPrivateKey = keyPair.getHexPrivateKey();
+        this.hexPublicKey = keyPair.getHexPublicKey();
+        logger.info("*** this.hexPrivateKey " + keyPair.getHexPrivateKey());
+        this.keyPair = KeyTool.convertHexedStringToKeyPair(this.hexPrivateKey, curveName);
+    }
+
+    public SDFSM2KeyPair(KeyPair javaKeyPair) {
+        super(javaKeyPair);
+        initSM2KeyPairObject();
+    }
+
+    protected SDFSM2KeyPair(CryptoResult result) {
+        super(result);
+        initSM2KeyPairObject();
+        logger.info(
+                "*** this.hexPrivateKey SDFSM2KeyPair(SDFCryptoResult sm2keyPairInfo) "
+                        + this.hexPrivateKey);
+        this.keyPair = KeyTool.convertHexedStringToKeyPair(this.hexPrivateKey, curveName);
+    }
+
+    protected SDFSM2KeyPair(long keyIndex, String password) {
```

Comment:
It seems the password is useless, please check again.

Reply:
It is usefull, thanks ^_^